src/bp/core/routers/admin/workspaces.ts

```diff
@@ -0,0 +1,131 @@
+import { Logger } from 'botpress/sdk'
+import { defaultPipelines } from 'common/defaults'
+import { CreateWorkspace } from 'common/typings'
+import { WorkspaceCreationSchema } from 'common/validation'
+import { ConfigProvider } from 'core/config/config-loader'
+import { InvalidOperationError } from 'core/services/auth/errors'
+import { BotService } from 'core/services/bot-service'
+import { WorkspaceService } from 'core/services/workspace-service'
+import { Router } from 'express'
+import Joi from 'joi'
+import _ from 'lodash'
+
+import { CustomRouter } from '../customRouter'
+import { NotFoundError } from '../errors'
+
+export class WorkspacesRouter extends CustomRouter {
+  constructor(
+    private logger: Logger,
+    private workspaceService: WorkspaceService,
+    private botService: BotService,
+    private configProvider: ConfigProvider
+  ) {
+    super('Workspace', logger, Router({ mergeParams: true }))
+    this.setupRoutes()
+  }
+
+  setupRoutes() {
+    const router = this.router
+    router.get(
+      '/',
+      this.asyncMiddleware(async (req, res) => {
+        res.send(await this.workspaceService.getWorkspaces())
+      })
+    )
+
+    router.post(
+      '/',
+      this.asyncMiddleware(async (req, res) => {
+        const { error } = Joi.validate(req.body, WorkspaceCreationSchema)
```

Comment:
You can't create one with custom, only none or botpress. But if you change workspaces.json manually and change it, then it is now custom.

Since you can change the pipeline afterwards through the UI, it will display custom, but will not change it